### PR TITLE
29.03.21 homework_git_27_28 done

### PR DESCRIPTION
**Pull request by EfimovEO**
github раньше не пользовался, надеюсь действую правильно.

### CODE REVIEW

File: Homework_git_27-28_EfimovEO.txt [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+[admin@andromeda lection_git_hw]$ cat ./homework 
+anything
+
+-------------------------------------------------------
+  6. Switch to "hw_git" branch again and create new file "temp_file" and commit it
+-------------------------------------------------------
+[admin@andromeda lection_git_hw]$ git checkout hw_git
+Switched to branch 'hw_git'
+[admin@andromeda lection_git_hw]$ touch temp_file
+[admin@andromeda lection_git_hw]$ git add temp_file && git commit ./temp_file -m "p.6 of hw"
```
Comment (SagurovA93) [inline Homework_git_27-28_EfimovEO.txt:66]: Use `git commit` directly to the file if you need to commit it directly.

This construction: `git add temp_file && git commit ./temp_file` should be replaced by `git commit`.

File: Homework_git_27-28_EfimovEO.txt [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  Make screenshots on each step below, pack them as tgz archive and attach it to your homework
+  3. Create group "hw_git"
+  4. Create two users: maintainer and developer
+  5. Add these users to the group and set them proper permissions in the group (maintainer – maintainer, developer – developer)
+  6. Create new project with any name
+  7. Create all branches for GitFlow in this project (you can create one feature and one release branch and don't create hotfix branch)
+  8. Protect master branch to allow only maintainers to merge into it, and restrict all to push there
+  9. Protect release branches by wildcard (release-* for example) and allow only maintainers to merge into it
+  10. Protect develop branch to allow everyone to create Merge Request into it
+  11. Allow anyone do anything in branches like "feature-*"
```
Comment (SagurovA93) [inline Homework_git_27-28_EfimovEO.txt:196]: According to this screenshot 
image


I have a question for you:
Can Maintainer push to the master branch with `push` and `push -f`?

**Review by SagurovA93 — CHANGES_REQUESTED** [review]
Good work! Could you review my comments in discussions and answer the questions/leave your comments, please. 

As additional questions I would like to ask the following questions:
1. What will happen when your local GitLab docker container will stop? Will it keep configurations and code?
2. How to setup root password before Gitlabb starts?

File: Homework_git_27-28_EfimovEO.txt (hunk 1 above)
Comment (EfimovEO) [reply Homework_git_27-28_EfimovEO.txt:66]: ok. i will do so next time

File: Homework_git_27-28_EfimovEO.txt (hunk 2 above)
Comment (EfimovEO) [reply Homework_git_27-28_EfimovEO.txt:196]: i think he can't

**Comment by EfimovEO** [comment]
> Good work! Could you review my comments in discussions and answer the questions/leave your comments, please.
> 
> As additional questions I would like to ask the following questions:
> 
>     1. What will happen when your local GitLab docker container will stop? Will it keep configurations and code?
> 
>     2. How to setup root password before Gitlabb starts?

1. When local Gitlab docker container will stop it will keep configurantions and code, because Gitlab writes them to the VOLUMES, which were created when container has started.

2. I didn't try but here is what i've found:
 define GITLAB_OMNIBUS_CONFIG as an environment variable in docker-compose file, like:
     environment:
      GITLAB_OMNIBUS_CONFIG: |
        gitlab_rails['initial_root_password'] = 'adminadmin'

File: Homework_git_27-28_EfimovEO.txt (hunk 2 above)
Comment (SagurovA93) [reply Homework_git_27-28_EfimovEO.txt:196]: Could you check it? :)

**Review by SagurovA93 — APPROVED** [review]

**Comment by SagurovA93** [comment]
Good to go

**Pull request merged**